### ISSUE TITLE
Component can receive new props with old results

### ISSUE THREAD

**Issue by sjoerdvisscher** [Jpunt/react-promise-container#3]
If a promiseContainer receives new props, it is passed on to the component immediately, while the result gets updated later.

So either executePromise should start with setting status to pending, or promiseContainer should pass the old props until the promises are resolved.

**Comment by Jpunt** [comment]
Can you give me an example of a use case? It's very common for props to change in places where promiseContainers live (without it being relevant to that promiseContainer), so I'm not sure if I'd want to break the regular render-flow of react here.. 🤔

**Comment by miracle2k** [comment]
In my case, I have a routable page that is rendered for /product/:name. If the user switches from one product directly to another, the promise would not be reloaded.

**Comment by sjoerdvisscher** [comment]
The specific case was a bug in blink: we got errors that the link component was missin an id value. It turned out that this happened in "beheer" while switching between different collections. After switching, props are updated, including a new idColumnName value. But the promise hadn't resolved yet, so we were still rendering the old results which don't have that specific id column.

**Comment by Jpunt** [comment]
Yeah, now that I've read your usecases I can recall some examples of something similar. I agree with @sjoerdvisscher that the state should flip back to pending and the promises should refresh (based on the new props). 

I don't think you'd always want this though, because there could be situations where props change and you don't need to update these promises. 

To do this, you can already use (the undocumented) `shouldPromiseRefresh`, but it won't set its state to pending (yet).

```js
function mapPromisesToProps(params) {
  return {
    post: getPost(ownProps.params.id),
  }
}

export default promiseContainer(mapPromisesToProps, {
  shouldPromiseRefresh: (ownProps, nextProps) => ownProps.params.id != nextProps.params.id,
})(ExampleComponent);
```

I'll get to flipping the state back to pending ASAP.
